src/main/scala/io/netty/handler/ssl/util/SelfSignedCert.scala

```diff
@@ -0,0 +1,30 @@
+package io.netty.handler.ssl.util
+
+import java.io.File
+import java.security.{KeyPairGenerator, SecureRandom}
+
+
+/**
+ * A container for a self-signed certificate and its private key.
+ *
+ * We use netty's code for generating self-signed certificates, but specialized to always
+ * use their bouncy castle generator, which uses SHA256 to sign the cert, instead of SHA1
+ * as used by their jdk certificate generator.
+ *
+ * We put this in the io.netty.handler.ssl.util package so that we can access the package
+ * protected BouncyCastleSelfSignedCertGenerator class.
+ */
+case class SelfSignedCert(certificate: File, privateKey: File)
```

Comment:
Ah ok, that makes it clearer thanks. 

I also came across http://stackoverflow.com/questions/2312881/what-is-the-difference-between-scalas-case-class-and-class) that made for interesting reading and fits into the view used here.